rudder-core/src/main/scala/com/normation/rudder/repository/jdbc/ExpectedReportsJdbcRepository.scala

```diff
@@ -110,95 +107,115 @@ class FindExpectedReportsJdbcRepository(
   import pgInClause._
   import doobie._
 
+
   /*
-   * Retrieve the list of node config ids
+   * Retrieve the expected reports by config version of the nodes.
+   *
+   * The property "returnedMap.keySet == nodeConfigIds" holds.
    */
-  override def getNodeConfigIdInfos(nodeIds: Set[NodeId]): Box[Map[NodeId, Option[Seq[NodeConfigIdInfo]]]] = {
-    getNodeConfigIdInfosIO(nodeIds).attempt.transact(xa).run
+  override def getExpectedReports(
+      nodeConfigIds: Set[NodeAndConfigId]
+  ): Box[Map[NodeAndConfigId, Option[NodeExpectedReports]]] = {
+
+    nodeConfigIds.toList match { //"in" param can't be empty
+      case Nil        => Full(Map())
+      case nodeAndIds =>
+
+        (for {
+          configs <- query[NodeExpectedReports](s"""
+                       select nodeid, nodeconfigid, begindate, enddate, configuration
+                       from nodeconfigurations
+                       where (nodeid, nodeconfigid) in (${nodeAndIds.map(x => s"(${x.nodeId.value},${x.version.value})").mkString(",")} )
+                     """).vector
+        }yield {
+          val configsMap = configs.map(x => (NodeAndConfigId(x.nodeId, x.nodeConfigId), x)).toMap
+          nodeConfigIds.map(id => (id, configsMap.get(id))).toMap
+        }).attempt.transact(xa).run
+    }
   }
 
+  /*
+   * Retrieve the current expected report for the list of nodes.
+   *
+   * The property "returnedMam.keySet = nodeIds" holds.
+   */
+  override def getCurrentNodeConfigurations(nodeIds: Set[NodeId]): Box[Map[NodeId, Option[NodeExpectedReports]]] = {
+    nodeIds.toList.toNel match {
+      case None => Full(Map())
+      case Some(ids) =>
+
+        implicit val idsParam = Param.many(ids)
+
+        (for {
+          configs <- sql"""
+                       select nodeid, nodeconfigid, begindate, enddate, configuration
+                       from nodeconfigurations
+                       where enddate is null and nodeid in (${ids: ids.type})
+                     """.query[\/[(NodeId, NodeConfigId, DateTime), NodeExpectedReports]].vector
+        } yield {
+          val configsMap = (configs.flatMap {
+            case -\/((id, c, _)) =>
+              logger.error(s"Error when deserializing JSON for expected node configuration ${id.value} (${c.value})")
+              None
+            case \/-(x) => Some((x.nodeId, x))
+          }).toMap
+          nodeIds.map(id => (id, configsMap.get(id))).toMap
+        }).attempt.transact(xa).run
+    }
+  }
 
-  /**
-   * This is the implementation of getNodeConfigIdInfos but without commiting the
-   * transaction so that it can be used in a sequence of queries.
+
+  override def findCurrentNodeIds(ruleId : RuleId) : Box[Set[NodeId]] = {
+    sql"""
+      select distinct nodeid from nodeconfigurations
+      where enddate is null and configuration like ${"%"+ruleId.value+"%"}
```

Comment:
Yeah... And it's a temporary solution (for 4.0, I think) until we have a "compliance" table, with compliance result by